### PR TITLE
Improve tag extraction

### PR DESCRIPTION
**Pull request by otbutz**
- [x] support for parsing short ton weight limits
- [x] move speed limit parsing into OSMValueExtractor
- [x] utility to check parsers using JSON data provided by https://taginfo.openstreetmap.org/

### CODE REVIEW

File: core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java [hunk 1]
```diff
@@ -222,20 +223,35 @@ public double getMaxSpeed() {
...
+        return !Double.isNaN(speed) && speed >= 0;
+    }
+
+    /**
+     * @return <i>true</i> if the given speed is not {@link Double#NaN} and is
+     *         bigger than zero
+     */
+    protected boolean isPositiveSpeed(double speed) {
+        return !Double.isNaN(speed) && speed > 0;
+    }
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java:254]: why is this necessary?

Comment (otbutz) [reply core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java:254]: Yeah i was thinking about the same. Wouldn't it be better to ignore 0 speeds completely?

https://overpass-turbo.eu/s/WwG

Most of those ways have an accompanying access tags which already prohibit routing.

**Comment by otbutz** [comment]
We should probably add support for parsing non-numerical maxheight values: https://wiki.openstreetmap.org/wiki/Key:maxheight#Non-numerical_values

Currently we miss out on a lot of valuable information:

* default: 18545
* none: 12264
* below_default: 3045
* unsigned: 136
* no_indications: 84

File: core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java (hunk 1 above)
Comment (karussell) [reply core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java:254]: Yes, your overpass query is convincing, seems to be a good idea to exclude those ways and assume this is a tagging problem. Or is it used for other purposes (e.g. for other speed tags) too?

Comment (otbutz) [reply core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java:254]: I couldn't find any reasonable uses of speed 0 for `maxspeed:forward`, `maxspeed:backward` and `maxspeed:motorcycle`. It should be safe to ignore it as a tagging error.

Related: https://github.com/graphhopper/graphhopper/issues/367

**Comment by otbutz** [comment]
IMHO it's better to ignore `default` and its equivalents and handle non existing limits later using spatial rules as we're able to properly differentiate between countries.

But i've the extended the extractor to deal with maxheight `below_default` returning a rather conservative height limit of 2m which should be enough for regular cars to pass and while blocking HGV traffic.

File: core/src/main/java/com/graphhopper/routing/ev/MaxSpeed.java [hunk 2]
```diff
@@ -33,6 +33,10 @@
      * The speed value used for road sections without known speed limit.
      */
     public static final double UNSET_SPEED = Double.POSITIVE_INFINITY;
+    /**
+     * The lowest speed value which can be stored.
+     */
+    public static final double MINIMUM_STOREABLE_SPEED = 5;
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/ev/MaxSpeed.java:39]: This is not that easy and dependent on how the EncodedValue for the speed is configured. So we can easily store 0.1km/h with enough bits...

File: core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java [hunk 3]
```diff
@@ -80,6 +86,12 @@ public static double stringToMeter(String value) {
         value = FEET_PATTERN.matcher(value).replaceAll("ft");
         if (isInvalidValue(value))
             return Double.NaN;
+        
+        // only used for maxheight
+        if ("below_default".equals(value)) {
+            return 2; // assume that cars are able to pass
+        }
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java:93]: I would not do this and instead ignore all these soft tags. IMO tagging a height with such a value is a tagging issue and we should not use it or even convert it into an explicit max height.

Comment (otbutz) [reply core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java:93]: `below_default` is an accepted and documented value. Projects like [StreetComplete](https://github.com/westnordost/StreetComplete) make use of it: https://github.com/westnordost/StreetComplete/blob/ed6ab1426f5c26dc4779c2c84118483a4a84a6eb/app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeight.kt#L68-L77

File: core/src/main/java/com/graphhopper/routing/ev/MaxSpeed.java (hunk 2 above)
Comment (otbutz) [reply core/src/main/java/com/graphhopper/routing/ev/MaxSpeed.java:39]: I'm not sure how (or if?) we should deal with unstoreable values. If we know that we can't store a certain value, we could use alternative tags or fallback values.

File: core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java (hunk 3 above)
Comment (karussell) [reply core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java:93]: Ok, yes. I can see it in the wiki. It is not really useful IMO and seems to be mostly used by German mappers. My problem with this approach is for GraphHopper itself: if we would do this then some limits have not the meaning they have in real world. So I would keep this for another issue.

File: core/src/main/java/com/graphhopper/routing/ev/MaxSpeed.java (hunk 2 above)
Comment (karussell) [reply core/src/main/java/com/graphhopper/routing/ev/MaxSpeed.java:39]: The tag extraction methods shouldn't care about it and always return precise numbers. Instead this issue can be handled in OSMMaxSpeedParser dependent on the configuration of the EncodedValue. As this is an important issue it is worth a separate issue.

**Comment by karussell** [comment]
Sorry, I completely missed this ... will merge now.

**Review by karussell — APPROVED** [review]
